parquet-encoding/src/main/java/org/apache/parquet/column/values/bitpacking/ByteBasedBitPackingEncoder.java

```diff
@@ -75,6 +79,10 @@ public void writeInt(int value) throws IOException {
       pack();
       if (packedPosition == slabSize) {
         slabs.add(BytesInput.from(packed));
+        totalFullSlabSize += slabSize;
+        if (slabSize < bitWidth * MAX_SLAB_SIZE_MULT) {
+          slabSize *= 2;
+        }
```

Comment:
This looks fine
@rdblue @isnotinvain @piyushnarang does it look good to you?
@JohnPJenkins did you check the performance impact of this?